cpp/proto/serializer_v2.cc

```diff
@@ -0,0 +1,108 @@
+/* -*- indent-tabs-mode: nil -*- */
+#include "proto/serializer_v2.h"
+
+#include <glog/logging.h>
+
+#include "util/status.h"
+#include "util/openssl_util.h"
+
+namespace rfc6962_bis {
+using util::StatusOr;
+using util::Status;
+
+namespace {
+const char kDerOIDTag = 0x06;
+}
+
+OID::OID() : oid_(nullptr) {
+}
+
+OID::OID(ASN1_OBJECT* oid) : oid_(oid) {
+  CHECK_NOTNULL(oid_);
+}
+
+OID::OID(const OID& other) : oid_(nullptr) {
+  if (other.oid_ != nullptr) {
+    oid_ = OBJ_dup(other.oid_);
+  }
+}
+
+OID::~OID() {
+  if (oid_ != nullptr) {
+    ASN1_OBJECT_free(oid_);
+  }
+}
+
+std::string OID::ToTagMissingDER() const {
+  if(oid_ == nullptr) {
+    // Uninitialized.
+    return "";
```

Comment:
I'm suspicious if this should be a StatusOr<string>. Is it only used to display values? If we did get an erroneous value we wouldn't be able to display it.